crates/sui-indexer-alt-framework/src/pipeline/sequential/committer.rs

```diff
@@ -189,6 +189,10 @@ where
                         continue;
                     }
 
+                    let Some(watermark) = watermark else {
```

Comment:
I initially had a question here about whether we should be skipping all the metrics updates below if we didn't have a `watermark`, because the first two were about `batch_rows` and `batch_checkpoint`. In fact this is fine, because `watermarks.is_none()` implies `batch_checkpoints == 0` so we would have skipped that because of the condition above, but maybe it's worth extending the doc comment above to explain that, something like:

> Conversely, if there is no watermark to be updated, we know there is no data to write out.